### PR TITLE
feat: add support for Passive Health Checks (Outlier Detection)

### LINKED ISSUE #2552: Add support for Passive Health Checks (Outlier Detection)
*Description*:

Add support for https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier

Relates to https://github.com/envoyproxy/gateway/issues/2511

### PR DESCRIPTION
**Pull request by yeedove**
**What this PR does / why we need it**:
add support for Passive Health Checks
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier
**Which issue(s) this PR fixes**:

Fixes #2552

### CODE REVIEW

File: api/v1alpha1/healthcheck_types.go [hunk 1]
```diff
@@ -13,6 +13,60 @@ type HealthCheck struct {
...
+
+// PassiveHealthCheck defines the configuration for passive health checks in the context of Envoy's Outlier Detection,
+// see https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier
+type PassiveHealthCheck struct {
+
+	// SplitExternalLocalOriginErrors enables splitting of errors between external and local origin.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	SplitExternalLocalOriginErrors bool `json:"splitExternalLocalOriginErrors,omitempty"`
```
Comment (shawnh2) [inline api/v1alpha1/healthcheck_types.go:30]: use ptr for optional type

File: internal/ir/xds.go [hunk 2]
```diff
@@ -1327,9 +1329,36 @@ type CircuitBreaker struct {
...
+	Passive *OutlierDetection `json:"passive,omitempty" yaml:"passive,omitempty"`
+}
+
+// OutlierDetection defines passive health check settings
+// +k8s:deepcopy-gen=true
+type OutlierDetection struct {
+	// Interval defines the time between passive health checks.
+	Interval *metav1.Duration `json:"interval,omitempty"`
+	// SplitExternalLocalOriginErrors enables splitting of errors between external and local origin.
+	SplitExternalLocalOriginErrors bool `json:"splitExternalLocalOriginErrors,omitempty" yaml:"splitExternalLocalOriginErrors,omitempty"`
```
Comment (shawnh2) [inline internal/ir/xds.go:1343]: ditto

**Review by shawnh2 — COMMENTED** [review]
add some comments, generally LGTM

**Comment by zirain** [comment]
let's fcous on API first, otherwise there may be a lot of waste your time.

**Comment by yeedove** [comment]
@zirain @shawnh2 done, thanks!

**Comment by arkodg** [comment]
thanks for adding this API @yeedove 
looks like the fields introduced are a 1:1 subset of https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#
@envoyproxy/gateway-maintainers should we introduce all these fields as part of the first iteration ?

**Review by arkodg — DISMISSED** [review]
LGTM thanks for introducing this !

**Review by shawnh2 — APPROVED** [review]
LGTM

**Review by zirain — APPROVED** [review]

**Comment by yeedove** [comment]
Resolved the conflict.

**Review by arkodg — APPROVED** [review]

**Pull request merged**